Sources/MongoSwift/APM.swift

```diff
@@ -14,6 +14,133 @@ public struct ConnectionId {
         }
         self.port = hostData.port
     }
+
+    /// Initializes a ConnectionId at localhost:27017, the default host/port.
+    internal init() {
+        self.host = "localhost"
+        self.port = 27017
+    }
+}
+
+/// The possible types for a server. The raw values correspond to the values libmongoc uses. 
+/// (We don't use these strings directly because Swift convention is to use lowercase for enums.)
+public enum ServerType: String {
+    case standalone = "Standalone"
+    case mongos = "Mongos"
+    case possiblePrimary = "PossiblePrimary"
+    case rsPrimary = "RSPrimary"
+    case rsSecondary = "RSSecondary"
+    case rsArbiter = "RSArbiter"
+    case rsOther = "RSOther"
+    case rsGhost = "RSGhost"
+    case unknown = "Unknown"
+}
+
+/// A struct describing a mongod or mongos process.
+public struct ServerDescription {
```

Comment:
libmongoc `mongoc_server_description_t`s also have an `id` available ([docs](http://mongoc.org/libmongoc/current/mongoc_server_description_id.html)), which isn't part of the spec here but we could add it... I'm not really sure what its purpose is though